client/my-sites/email/gsuite-add-users/index.jsx

```diff
@@ -174,13 +174,15 @@ class GSuiteAddUsers extends Component {
 	}
 
 	goToEmail = () => {
-		page(
-			emailManagement(
-				this.props.selectedSite.slug,
-				this.props.selectedDomainName,
-				this.props.currentRoute
-			)
-		);
+		const { currentRoute, selectedDomainName, selectedSite, source } = this.props;
+
+		let url = emailManagement( selectedSite.slug, selectedDomainName, currentRoute );
+
+		if ( source === 'inbox' ) {
+			url = emailManagementInbox( selectedSite.slug );
+		}
+
+		page( url );
```

Comment:
This is a good initiative, but with the very recent completion of the "unify mailboxes" project, this component is no longer in use.

You should rebase this branch and move this implementation to the `goToEmail` function in `client/my-sites/email/add-mailboxes/index.tsx`.